### PR TITLE
Add view schema and validate that view YAML file matches during discovery

### PR DESCRIPTION
**Pull request by rose-huang**
## Description
- Add view_schema  following the fields in https://airplane.slab.com/posts/2022-06-07-views-definition-umob2ewz
- unmarshal the .app.yaml file and validate it matches the schema
- renamed AppConfig type to ViewConfig 

## Test plan
Ran the discover_test
Built the CLI by replacing 
`replace github.com/airplanedev/lib => /Users/rosehuang/airplanedev/lib`

updated the names of AppConfig -> ViewConfig and AppConfig.Slug -> AppConfig.Def.Slug and ran a local deploy (`bin/airplane deploy ../lib/pkg/examples/app/simple --dev --host=api.airplane.so:5000`) and it started properly

Tested a schema without the required fields and with extra field and confirmed it errored

### CODE REVIEW

File: pkg/deploy/discover/discover.go [hunk 1]
```diff
@@ -38,15 +38,14 @@ func (c TaskConfig) GetSource() ConfigSource {
...
-type AppConfig struct {
-	ID         string
-	Root       string
-	Entrypoint string
-	Slug       string
-	Source     ConfigSource
+type ViewConfig struct {
+	ID     string
+	Root   string
+	Def    definitions.ViewDefinition
```
Comment (rose-huang) [inline pkg/deploy/discover/discover.go:44]: In TaskConfig, Def is a `definitions.DefinitionInterface` with functions to get the slug, name, etc from definition so I can make this like that as well but idk if it's necessary?

File: pkg/deploy/discover/app_defn_discover.go [hunk 2]
```diff
@@ -37,42 +30,40 @@ func (dd *AppDefnDiscoverer) GetAppConfig(ctx context.Context, file string) (*Ap
...
 	if err != nil {
 		var merr *api.AppMissingError
 		if !errors.As(err, &merr) {
-			return nil, errors.Wrap(err, "unable to get app")
+			return nil, errors.Wrap(err, "unable to get view")
 		}
 		// TODO offer to create the app.
 		if dd.Logger != nil {
-			dd.Logger.Warning(`App with slug %s does not exist, skipping deploy.`, d.Slug)
+			dd.Logger.Warning(`View with slug %s does not exist, skipping deploy. :(`, d.Slug)
```
Comment (leeweisberger) [inline pkg/deploy/discover/app_defn_discover.go:61]: lol maybe remove the frowny face

File: pkg/deploy/discover/discover.go (hunk 1 above)
Comment (leeweisberger) [reply pkg/deploy/discover/discover.go:44]: Oh man let's not do that :)

File: pkg/deploy/discover/discover.go [hunk 3]
```diff
@@ -124,9 +123,9 @@ func (d *Discoverer) Discover(ctx context.Context, paths ...string) ([]TaskConfi
 				taskConfigsBySlug[slug] = append(taskConfigsBySlug[slug], tc)
 			}
 			for _, ac := range nestedAppConfigs {
-				slug := ac.Slug
+				slug := ac.Def.Slug
```
Comment (leeweisberger) [inline pkg/deploy/discover/discover.go:126]: nit - change ac -> vc

File: pkg/deploy/discover/discover_test.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func TestDiscover(t *testing.T) {
 		existingApps        map[string]api.App
 		expectedErr         bool
 		expectedTaskConfigs []TaskConfig
-		expectedAppConfigs  []AppConfig
+		expectedAppConfigs  []ViewConfig
```
Comment (leeweisberger) [inline pkg/deploy/discover/discover_test.go:25]: expectedViewConfigs

File: pkg/deploy/taskdir/definitions/view_schema.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    "name": {
+      "description": "A human-readable name for your view.",
+      "type": "string"
+    },
+    "description": {
+      "description": "A human-readable description for your view.",
+      "type": "string"
+    },
+    "entrypoint": {
```
Comment (leeweisberger) [inline pkg/deploy/taskdir/definitions/view_schema.json:21]: Let's make this optional and default it to the current working directory. You can either do in this pr or a follow up

**Review by leeweisberger — APPROVED** [review]

**Pull request merged**